guide/src/ref/packet-log.md

```diff
@@ -1,37 +1,167 @@
+Rattan (e.g., UDP flows).
 
 ## Usage
 
-Using the `--packet-log` option in the CLI will enable the packet log feature and specify the file to store the records. The flow description file will be stored in the same directory with the name `<packet_log>.flows`, where `<packet_log>` is the name of the packet log file.
+### Record a Rattan packet Log
 
-For example, you can run the following command to enable packet logging:
+Two options are involved: `--packet-log` and `--packet-log-mode` . Example usages are:
 
 ```bash
-rattan link --packet-log packet.log
+rattan link --packet-log packet.rtl --packet-log-mode compact-tcp
+rattan run --config CONFIG_FILE --packet-log packet.rtl --packet-log-mode raw-ip
 ```
 
-This will create a file named `packet.log` in the current directory, which will contain the compressed packet log. The flow description file will be named `packet.log.flows`.
+The possible values for `--packet-log-mode` are :
+
+- `compact-tcp` (as default, if not specified). In this mode, some fields from TCP and IP headers are recorded.
+- `raw-ip`. In this mode, the raw L3 and L4 headers are recorded.
+- `raw-tcp`. In this mode, the raw L4 headers are recorded.
+
+This will create a file named `packet.rtl` in the current directory, which will contain the compressed packet log.
+The flow description file will be named `packet.flows`. Also, if any raw headers are recorded, a `packet.raw` will be created.
+
+### Convert a Rattan packet log
+
+#### Rust subcommand
+
+To convert a Rattan packet log to `.pcapng` file, which can be viewed with Wireshark or other tools, we provide
+a subcommand `convert`.
 
-We also provide a python script [scripts/log_converter.py](https://github.com/stack-rs/rattan/blob/main/scripts/log_converter.py) to convert the packet log file (along with the flow description file) to some pcapng files, which can be viewed with Wireshark or other tools. You can run the script like this:
+Example usage:
 
 ```bash
-python3 scripts/log_converter.py packet.log converted.pcapng
+rattan convert ./packet.rtl [./output]
 ```
 
-This will read `packet.log` and `packet.log.flows`, and generate a pcapng file for each end of Rattan (identified by IP address), with the name `converted_<ip>.pcapng` (e.g. `converted_10_1_1_1.pcapng`).
+This expects the following files exist, as input:
 
-Note that some fields are artificially constructed in the pcapng file, as the compressed packet log only stores partial information. Artificial fields include:
+- `./packet.rtl`
+- `./packet.flows`
+- `./packet.raw` (optional, only if the packet log was recorded in `raw-tcp` or `raw-ip` mode).
+
+And it will Generated the following files, as output, if the address of the two ends of the emulated link
+of Rattan was `10.1.1.1` and `10.2.1.1`":
+
+- `./output_10_1_1_1.pcapng`
+- `./output_10_2_1_1.pcapng`
+
+If the second argumnet was not provided(the output prefix), the default value for it is the first argument (the path
```

Comment:
Also, I wonder if it is possible that I specify a directory-like prefix. Suppose I run:

```bash
rattan convert ./packet.rtl ./output/layer3
```

What will happen? Will it produce files to ./output/layer3_10_1_1_1.pcapng ? Whether this is correct or not, we should note this in our documentationi.